### PR TITLE
chore: upgrade rust to 1.82

### PR DESCRIPTION
**Pull request by satyarohith**

### CODE REVIEW

File: runtime/permissions/prompter.rs [hunk 1]
```diff
@@ -269,7 +269,7 @@ fn get_stdin_metadata() -> std::io::Result<std::fs::Metadata> {
   unsafe {
     let stdin = std::fs::File::from_raw_fd(0);
     let metadata = stdin.metadata().unwrap();
-    stdin.into_raw_fd();
+    let _ = stdin.into_raw_fd();
```
Comment (bartlomieju) [inline runtime/permissions/prompter.rs:272]: This looks fishy, why does it now require `let _ = `?

File: cli/tools/info.rs [hunk 2]
```diff
@@ -527,15 +527,15 @@ impl<'a> GraphDisplayContext<'a> {
 
   fn build_module_info(&mut self, module: &Module, type_dep: bool) -> TreeNode {
     enum PackageOrSpecifier {
-      Package(NpmResolutionPackage),
+      Package(Box<NpmResolutionPackage>),
```
Comment (bartlomieju) [inline cli/tools/info.rs:530]: I assume this is now flagged by linter because of enum variant size diff?

File: tests/napi/src/async.rs [hunk 3]
```diff
@@ -95,7 +95,7 @@ extern "C" fn test_async_work(
   ));
   let mut baton = unsafe { Box::from_raw(baton_ptr as *mut Baton) };
   baton.task = async_work;
-  Box::into_raw(baton);
+  let _ = Box::into_raw(baton);
```
Comment (bartlomieju) [inline tests/napi/src/async.rs:98]: Ditto, what's getting flagged here?

File: cli/tools/info.rs (hunk 2 above)
Comment (satyarohith) [reply cli/tools/info.rs:530]: yes

File: tests/napi/src/async.rs (hunk 3 above)
Comment (satyarohith) [reply tests/napi/src/async.rs:98]: I'm getting clippy lint error for unused return value

**Comment by satyarohith** [comment]
Closing this in preference to https://github.com/denoland/deno/pull/26615

**Pull request closed without merging**